interactive_engine/executor/ir/graph_proxy/src/apis/router.rs

```diff
@@ -16,10 +16,12 @@
 use crate::apis::ID;
 use crate::GraphProxyResult;
 
-pub trait Partitioner: Send + Sync + 'static {
-    /// Given the element id and job_workers (number of worker per server),
-    /// return the id of worker that is going to process
-    fn get_partition(&self, id: &ID, job_workers: usize) -> GraphProxyResult<u64>;
-    /// Get the local partition list that is going to be processed by current server.
-    fn get_local_partitions(&self) -> GraphProxyResult<Vec<u32>>;
+/// A router used to route the data to the destination worker.
+/// Specifically, given the graph element id to query,
+/// it firstly route the data to the server (aka. process) which is able to access the queried data from graph storage,
+/// and then pick a worker (aka. thread) in that server to do the query.
+pub trait Router: Send + Sync + 'static {
+    /// Given the element id and job_workers (number of workers per server),
+    /// return the worker id that is going to do the query.
+    fn route(&self, id: &ID, job_workers: usize) -> GraphProxyResult<u64>;
```

Comment:
I have the following suggestions for this api:

- Design a structure, such like `ClusterInfo` here, to include how the whole cluster is managed, including how many servers (we actually need this, currently, we simply leverage a pegasus static function to obtain this information, which couples implementation with this api) in the cluster and how many workers in a server. The ClusterInfo can be initiated while starting the server, and then pass into this function as a reference. 
- Make the item of `id` as an abstraction of `RouteData`, where `RouteData` implements the function `get_route_id()`. Then we probably can directly implement `Vertex` and `Edge` (or even `ID`) as `RouteData`. `ID` is a little bit ambiguous given we actually want to route any data, not just `ID`.
-  Make `u64` a type of `WorkerId`.  
- I think Graph partition information should also be aware to the `Router`, no?
- Comment this structure as:
```
While processing 
```
A `Router` is used to route the data to the destination worker so that it can be properly processed, especially
when the underlying data has been partitioned across the cluster. Given the partition information as well as how our cluster is managed (by `ClusterInfo`) and co-located with the graph data, we can implement the corresponding `route` function to guide the system to transfer the data to a proper destination worker. 

For example, suppose our computer server contains 10 servers, each further forking 10 workers for processing queries. In addition, the graph is partitioned into these 10 servers by the following strategy: vertex of give ID is placed in the server with id i (0 to 9) i given ID % 10 == i, the vertex's adjacent edges are also placed with the vertex. Then the router can decide which worker should process the vertex of ID 25534 as follows:
- It first do `25534 % 10 == 4`, which means it must be routed to the 4-th server.
- Any worker in the 4-th server can process the vertex. Thus it randomly picks a worker, saying 5-th worker, which has ID 4 * 10 + 5 = 45.
- Then 45-th worker will be returned for routing this vertex.
```